### ISSUE TITLE
Recognize any barcode

### ISSUE THREAD

**Issue by swellfr** [mmaitre314/VideoEffect#10]
I've just tried your lib on my phone to do real time QR code analysis and it's very impressive how fast it is.

Would it be difficult to recognize all zxing barcodes? I tried on code 128 but it didn't work.

Thanks

**Comment by mmaitre314** [comment]
I haven't tried but that should be possible. What did you use as DecodingOptions? Maybe the quality of the images is too low for ZXing. ZXing has difficulties with blurred images (happens often on phones without continuous auto-focus like the 520). In the QR code test app there is a button to snap images to JPEG files and check what they look like: https://github.com/mmaitre314/VideoEffect/blob/master/VideoEffects/QrCodeDetector/QrCodeDetector.Shared/MainPage.xaml.cs (see beginning of AnalyzeBitmap()).

**Comment by swellfr** [comment]
I tried with only code 128 decode options, and with some big barcode images on Google image to avoid blurred images.

It didn't work for me.

**Comment by mmaitre314** [comment]
Ok, will try to see if I can figure out something.

**Comment by AmrTealeb** [comment]
I have the same problem
I can use the QR test app to detect QR Codes but not Code_128 or Code_39

**Comment by mmaitre314** [comment]
I added some tests in commit 75988cfb974386d5689da99a10e1eb6931bcf7ba to see how ZXing and Lumia Bitmap worked together. It looks like the problem is due to the bitmap stride (at least that's my best theory right now).

This piece of code in the QrCodeDetector test app is not fully correct:

```C#
Result result = m_reader.Decode( 
    bitmap.Buffers[0].Buffer.ToArray(), 
    (int)bitmap.Dimensions.Width, 
    (int)bitmap.Dimensions.Height, 
    BitmapFormat.Gray8 
    ); 
```
ZXing only takes {width, height} as input while {width, height, stride} is actually needed to describe the bitmap data layout (potential padding at the end of each line).

Getting the stride wrong transforms vertical lines into diagonal ones. The QR code detector is likely more forgiving than the Code 128 one when that happens. That could also explain why QR code detection was a bit flaky in the test app.

Two workarounds that worked better in the tests:
1. Copy the Lumia bitmap from Yuv420Sp to Bgra8888 before passing it to ZXing. It looks like there is no padding there (at least with the resolutions I tried)
2. Replace "(int)bitmap.Dimensions.Width" by "(int)bitmap.Buffers[0].Pitch" in the Decode() call. The garbage pixels the decoder will process at the end of each line shouldn't impact code detection.
Option 2. should be more effective.
